core/src/main/java/org/apache/iceberg/view/ViewMetadataParser.java

```diff
@@ -0,0 +1,227 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SchemaParser;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.io.OutputFile;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.util.JsonUtil;
+import org.apache.iceberg.util.PropertyUtil;
+
+class ViewMetadataParser {
+
+  static final String FORMAT_VERSION = "format-version";
+  static final String LOCATION = "location";
+  static final String CURRENT_VERSION_ID = "current-version-id";
+  static final String VERSIONS = "versions";
+  static final String VERSION_LOG = "version-log";
+  static final String PROPERTIES = "properties";
+  static final String SCHEMAS = "schemas";
+  static final String CURRENT_SCHEMA_ID = "current-schema-id";
+
+  public static void overwrite(ViewMetadata metadata, OutputFile outputFile) {
+    internalWrite(metadata, outputFile, true);
+  }
+
+  public static void write(ViewMetadata metadata, OutputFile outputFile) {
+    internalWrite(metadata, outputFile, false);
+  }
+
+  public static void toJson(ViewMetadata metadata, JsonGenerator generator) throws IOException {
+    generator.writeStartObject();
+
+    generator.writeNumberField(FORMAT_VERSION, metadata.formatVersion());
+    generator.writeStringField(LOCATION, metadata.location());
+
+    JsonUtil.writeStringMap(PROPERTIES, metadata.properties(), generator);
+    generator.writeNumberField(CURRENT_SCHEMA_ID, metadata.currentSchemaId());
+    generator.writeArrayFieldStart(SCHEMAS);
+    for (Schema schema : metadata.schemas()) {
+      SchemaParser.toJson(schema, generator);
+    }
+
+    generator.writeEndArray();
+
+    generator.writeNumberField(CURRENT_VERSION_ID, metadata.currentVersionId());
+    generator.writeArrayFieldStart(VERSIONS);
+    for (ViewVersion version : metadata.versions()) {
+      ViewVersionParser.toJson(version, generator);
+    }
+
+    generator.writeEndArray();
+
+    generator.writeArrayFieldStart(VERSION_LOG);
+    for (ViewHistoryEntry viewHistoryEntry : metadata.history()) {
+      ViewHistoryEntryParser.toJson(viewHistoryEntry, generator);
+    }
+
+    generator.writeEndArray();
+    generator.writeEndObject();
+  }
+
+  static String toJson(ViewMetadata viewMetadata) {
```

Comment:
nit: maybe move this before the other `toJson(..)` method

Reply:
why this is not public?

Reply:
Jack makes a good point. A few of these should be reviewed. In addition to this, which reasonably should be public, I think that `toJson(ViewMetadata, JsonGenerator)` should not be public until we need it in another package.